pkg/autoscaler/aggregation/max/window.go

```diff
@@ -47,13 +47,12 @@ func (m *window) Record(index int, v float64) {
 	// An element that's lower than the new element can never influence the
 	// maximum again, because the new element is both larger _and_ more
 	// recent than it.
-	for l := m.length - 1; l >= 0; l-- {
-		// Search backwards because that way we can delete by just decrementing length.
-		// The elements are guaranteed to be in descending order as described in Step Three.
-		if v >= m.maxima[m.index(m.first+l)].value {
-			m.length--
-		} else {
-			// The elements are sorted, no point continuing.
+
+	// Search backwards because that way we can delete by just decrementing length.
+	// The elements are guaranteed to be in descending order as described in Step Three.
+	for ; m.length > 0; m.length-- {
+		// The elements are sorted, no point continuing.
```

Comment:
can we move this comment one line down so it's above the `break` since that's what it relates to? apart from that lgtm! 

(alternatively this could stay where it is but say sthg like "The elements are sorted, so continue until we find one that's lower than the new element").

Reply:
Done!